create-group4template.groovy

```diff
@@ -0,0 +1,157 @@
+    println "[ERROR]: Parameter pTemplate: `$pTemplate` does not exits in this instance"
+    processingFlag = false
+} else {
+    if (jenkins.getItemByFullName(pTemplate) instanceof Model==false){
+        println "[ERROR]: Parameter pTemplate: `$pTemplate` is not a Job Template type"
+        if (processingFlag) processingFlag = false
+    } else {
+        //println "[DEBUG]: Job Template `$pTemplate` has been found"
+    }
+}
+
+// 4.- ROLE_JOB_READ should be existing at contains Job:Read permission
+
+for (rol in roles.getRoles()){
+    //println "[DEBUG]: rol.id:" + rol.id
+    if((rol.id).equals(pRol)==true){
+        rolFoundFlag=true
+        //rolObj2assign = rol
+        for (permis in rol.getPermissionProxyIds()){
+            if(permis.equals(pPermission)==true){
+                permissionFoundFlag = true
+                break;
+            }
+        }
+    }
+}
+if (!rolFoundFlag){
+    println "[ERROR]: Parameter pRol: `$pRol` does not exist in this instance"
+    if (processingFlag) processingFlag = false
+} else {
+    if (!permissionFoundFlag){
+        println "[ERROR]: Parameter pPermission: `$pPermission` is not included into the Permission List of pRol: `$pRol`"
+        if (processingFlag) processingFlag = false
+    }
+}
+
+// 5.- There should not be an existing Group (pGroup) at Template (pTemplate) level
+
+groupIterator = groupsContainers
+for (gc2 in groupIterator) {
+    // nodeString has been build following same structure that the key of the Map groupsContainers
+    if ((gc2.key).equals(nodeString)){
+        // Group Container where changes should be applied. Node of pTemplate
+        gc2update = gc2.value
+        for (g in gc2.value.groups) {
+            if ((g.name).equals(pGroup)){
+                println "[ERROR]: There is an exiting group '$pGroup'in the '$pTemplate' of this instance"
+                if (processingFlag) processingFlag = false
+            }
+        }
+    }
+}
+
+// 6.- All users in pUsers should be existing in Jenkins
+
+for (u in pUsers) {
+    println "[DEBUG]:" + hudson.model.User.get(u)
+    //Checking if the user exist on Jenkins own's database
+    if(hudson.model.User.get(u,false,null)==null){
+        println "[ERROR]: User '" + u + "' is not included into Jenkins Own's database of this instance"
+        if (processingFlag) processingFlag = false
+    }
+}
+
+//----------------------
+//Processing
+//----------------------
+
+if (processingFlag) {
+    println "[DEBUG]: Strating to process it..."
+    //Creating group
+    g2create = new Group (gc2update, pGroup)
+    //Adding members
+    g2create.setMembers(pUsers)
+    def List<RoleAssignment> assignments = new ArrayList<RoleAssignment>()
+    def rolAssigmment = new RoleAssignment(pRol)
+    assignments.add(rolAssigmment)
+    //Adding roles Group#setRoleAssignments(List<RoleAssignment> roleAssignments)
+    g2create.setRoleAssignments(assignments)
+    g2create.save()
+    jenkins.save()
```

Comment:
The changes are not saved. Which method should I use for serialization at this point to make changes permanent?